docs/examples/pvfleets-qa-pipeline/pvfleets-irradiance-qa.py

```diff
@@ -0,0 +1,385 @@
+# CPD-estimated time shift series.
+midday_diff_series.plot()
+time_shift_series.plot()
+plt.title("Midday Difference Time Shift Series")
+plt.xlabel("Date")
+plt.ylabel("Midday Difference (Modeled-Measured), Minutes")
+plt.tight_layout()
+plt.show()
+
+# Plot the heatmap of the irradiance time series
+plt.figure()
+# Get time of day from the associated datetime column
+time_of_day = pd.Series(time_series.index.hour +
+                        time_series.index.minute/60,
+                        index=time_series.index)
+# Pivot the dataframe
+dataframe = pd.DataFrame(pd.concat([time_series, time_of_day], axis=1))
+dataframe.columns = ["values", 'time_of_day']
+dataframe = dataframe.dropna()
+dataframe_pivoted = dataframe.pivot_table(index='time_of_day',
+                                          columns=dataframe.index.date,
+                                          values="values")
+plt.pcolormesh(dataframe_pivoted.columns,
+               dataframe_pivoted.index,
+               dataframe_pivoted,
+               shading='auto')
+plt.ylabel('Time of day [0-24]')
+plt.xlabel('Date')
+plt.xticks(rotation=60)
+plt.title('Post-Correction Heatmap, Time of Day')
+plt.colorbar()
+plt.tight_layout()
+plt.show()
+
+# %%
+# Next, we check the time series for any abrupt data shifts. We take the
+# longest continuous part of the time series that is free of data shifts.
+# We use :py:func:`pvanalytics.quality.data_shifts.detect_data_shifts` to
+# detect data shifts in the time series.
+
+# Resample the time series to daily mean
+time_series_daily = time_series.resample('D').mean()
+data_shift_start_date, data_shift_end_date = \
+    ds.get_longest_shift_segment_dates(time_series_daily)
+data_shift_period_length = (data_shift_end_date - data_shift_start_date).days
+
+# Get the number of shift dates
+data_shift_mask = ds.detect_data_shifts(time_series_daily)
+# Get the shift dates
+shift_dates = list(time_series_daily[data_shift_mask].index)
+if len(shift_dates) > 0:
+    shift_found = True
+else:
+    shift_found = False
+
+# Visualize the time shifts for the daily time series
+print("Shift Found:", shift_found)
+edges = [time_series_daily.index[0]] + \
+    shift_dates + [time_series_daily.index[-1]]
+fig, ax = plt.subplots()
+for (st, ed) in zip(edges[:-1], edges[1:]):
+    ax.plot(time_series_daily.loc[st:ed])
+plt.title("Daily Time Series Labeled for Data Shifts")
+plt.xlabel("Date")
+plt.ylabel("Mean Daily Irradiance (W/m^2)")
+plt.tight_layout()
+plt.show()
+
+# %%
+# We filter the time series to only include the longest
+# shift-free period.
+
+# Filter the time series to only include the longest shift-free period
+time_series = time_series[
+    (time_series.index >= data_shift_start_date.tz_convert(
+        time_series.index.tz)) &
+    (time_series.index <= data_shift_end_date.tz_convert(
+        time_series.index.tz))]
+
+time_series = time_series.asfreq(data_freq)
+
+
+# %%
+# Display the final irradiance time series, post-QA filtering.
```

Comment:
This plot isn't rendering

Reply:
That plot still isn't rendering, idk why not.

Reply:
Looks like it's rendering now, not sure what was going on with it: 
![image](https://github.com/pvlib/pvanalytics/assets/70228568/d66a1562-81bb-4378-8834-47089452c8a7)